src/basilisp/core.lpy

```diff
@@ -547,6 +547,17 @@
     v
     (.-name v)))
 
+(defn tagged-literal
+  "Construct a data representation of a tagged literal from a
+  tag symbol and a form."
+  [tag form]
+  (basilisp.lang.tagged/tagged-literal tag form))
+
+(defn tagged-literal?
```

Comment:
```suggestion
(defn ^:inline tagged-literal
  "Construct a data representation of a tagged literal from a
  tag symbol and a form."
  [tag form]
  (basilisp.lang.tagged/tagged-literal tag form))

(defn ^:inline tagged-literal?
```

Minor but since you added `basilisp.lang.tagged` to the auto-import list, we can just inline these calls.